Userland/xargs.cpp

```diff
@@ -0,0 +1,276 @@
+        if (!replace_view.is_empty()) {
+            if (auto first_index = arg.find_first_of(replace_view); first_index.has_value()) {
+                some_argument_has_replacement = true;
+                while (first_index.has_value()) {
+                    auto current_index = first_index.value();
+                    auto substring_until_replacement = arg.substring_view(previous_index, current_index);
+                    parts.append(substring_until_replacement);
+                    previous_index = current_index;
+                    first_index = arg.substring_view(current_index + replace_view.length(), arg.length() - current_index - replace_view.length()).find_first_of(replace_view);
+                }
+                parts.append(arg.substring_view(previous_index + replace_view.length(), arg.length() - previous_index - replace_view.length()));
+            } else {
+                parts.append(arg);
+            }
+        } else {
+            parts.append(arg);
+        }
+        argument_parts.append(move(parts));
+        ++index;
+    }
+
+    // Append a '{}' at the end if no arguments have replacements.
+    if (!some_argument_has_replacement) {
+        Vector<StringView> parts;
+        parts.append("");
+        parts.append("");
+        argument_parts.append(move(parts));
+    }
+
+    FILE* fp = stdin;
+    bool is_stdin = true;
+
+    if (StringView { "-" } != file_to_read) {
+        // A file was specified, try to open it.
+        fp = fopen(file_to_read, "r");
+        if (!fp) {
+            perror("fopen");
+            return 1;
+        }
+        is_stdin = false;
+    }
+
+    char* line = nullptr;
+    size_t buffer_size = 0;
+
+    StringBuilder builder;
+    Vector<char*> child_argv;
+
+    int devnull_fd = 0;
+
+    if (is_stdin) {
+        devnull_fd = open("/dev/null", O_RDONLY | O_CLOEXEC);
+        if (devnull_fd < 0) {
+            perror("open");
+            return 1;
+        }
+    }
+
+    bool fail = false;
+    int lines_used_for_this_command = 0;
+    char* last_line = nullptr;
+    size_t last_line_size = 0;
+    size_t total_command_length = 0;
+
+    for (;;) {
+        line = nullptr;
+        buffer_size = 0;
+
+        ssize_t line_size;
+
+        if (last_line) {
+            line_size = last_line_size;
+            line = last_line;
+            last_line = nullptr;
+        } else {
+            line_size = getdelim(&line, &buffer_size, entry_separator, fp);
+        }
+
+        if (line_size < 0) {
+            if (errno != 0) {
+                perror("getdelim");
+                fail = true;
+            }
+            break;
+        }
+
+        last_line = line;
+        last_line_size = line_size;
+
+        if (line_size == 0) {
+            free(line);
+            last_line = nullptr;
+            continue;
+        }
+
+        ++lines_used_for_this_command;
+
+        if (line_size > 0) {
+            while (line[line_size - 1] == '\n')
+                --line_size;
```

Comment:
This will underflow if the whole line only consists of `\n`s

Reply:
Hmmm...that should actually check for only one newline.